### DISCUSSION TITLE
Newbee need help

### DISCUSSION THREAD

**Discussion by HolgerM1** [weltenwort/frigate-synology-dsm7#91] [Q&A]
Hi!
Right now I'm using my Synology NAS with the surveillance station and struggling with motion detection. That's the reason for me to give frigate a try, unfortunately I'm a noob in using docker.
I want to install frigate with coral on my DS 1817+ running DSM7.1.1
I read the readme on frigate-synology-dsm7 multiple times but in fact I even don't know whether to choose the installation with or without git (honestly I don't know what git means)
Is there any chance to handle that for me and could somebody give me a hint on how to start?!
Docker is already installed on the NAS as I was playing around with a DMS Software (ecoDMS) as docker installation. That worked fine.

Thanks for any help in advance
Holger

**Comment by weltenwort** [comment]
Hi @HolgerM1 :wave:

indeed, the instructions make quite a few assumptions about the familiarity with software development and system administration. Let me try to expand on them. If they work for you I can add them to the docs to help others that are in a similar situation.

**Docker**: Frigate is delivered as a container image, which is essentially a snapshot of the user land of an operating system (linux in this case). It contains the file system with the application pre-installed (frigate in this case) and some metadata. In order to run docker containers on DSM we need the "Docker" package to be installed via the Package Center. That also brings with it the `docker-compose` command-line utility, which we'll use to start the container. We use that, because the DSM Docker UI doesn't expose all necessary options that we need to set when starting the frigate container (e.g. granting access to the USB sub-system).

**SSH and root access to the Synology device**: In order to run `docker-compose` we need to be able to open a terminal on the system and acquire `root` privileges to communicate with the Docker Engine. How exactly we achieve that can vary a bit, but it most likely involves enabling the SSH service and creating a user with admin group privileges to log in with.

Unfortunately there's little I can do to make the configuration of frigate itself easier. You'll have to edit its configuration YAML file to suit your needs using the official frigate documentation.

Does that help somewhat?

**Reply by HolgerM1** [reply]
Thank you for the response. So basically if I already have ssh root access to the NAS and docker already installed, I can just follow the instructions you gave?! Or do I have to install frigate docker image on Synology in advance. As you obviously can see, my experience is very limited, sorry for that.

**Reply by weltenwort** [reply]
No worries, your questions are totally valid.

With ssh access and docker you should be prepared to follow the "without git" steps. You only need to download the `docker-compose.yml` file as the instructions say, because it refers to a prebuilt image I provide that docker will automatically download the first time you create the container. That's also why the first start might take a bit longer than any subsequent restarts.

In regard to upgrading to newer versions: You'll have to re-download (or edit) the compose file such that it refers to the newer version tag of the file when you want to update to a new release in the future. Then docker will download the new container image (identified by the new version tag) when you run `sudo docker-compose up --detach --force-recreate` again.

**Comment by HolgerM1** [comment]
Dear Felix! Thank you so much for having the patients to help a noob getting started. I really appreciate that! I think, I'll give it a try. As soon as I finished my other project.  Parallel on the issue with installing frigate I'm right know trying to get frigate running on an Khadas Vim2 max, which is definitely not easier...

**Comment by weltenwort** [comment]
That looks like a neat little board. Good luck and let us know how it works out. :four_leaf_clover:

**Comment by Floriszz** [comment]
Hi @weltenwort , I myself count me to the noob with little to no knowledge of cli and ssh. Also I prefer not to hack too much into the security of the Synology NAS. So prefer SSH access to off, as I don't have the knowledge about it and know this can be potentially dangerous vulnerability if configured wrong.
So I figured out a way without it. Learning from how it is done for other docker containers through the DSM task scheduler.
I see in your comment that you suggest to add it to the documentation. I tried to add/suggest it, through github, to add to: https://docs.frigate.video/frigate/installation/ under Docker under 'if you can't use docker compose' , but it does not show up for now.
a working minimal set looks like:
docker run -d
--name frigate
--restart=unless-stopped
--mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000
--device /dev/dri/renderD128
--shm-size=64m
--privileged
-v /path/to/your/storage:/media/frigate
-v /path/to/your/config.yml:/config/config.yml:ro
-v /path/to/your/db:/db
-v /etc/localtime:/etc/localtime:ro
-v /dev/bus/usb:/dev/bus/usb
-e FRIGATE_RTSP_PASSWORD='password'
-p 5055:5000
-p 1935:1935
blakeblackshear/frigate:stable

This is how it looks like in DSM task scheduler. By stopping and removing the container and image through the default DSM Docker app, you can even do updates by just running the task scheduler task. 
![image](https://user-images.githubusercontent.com/19168937/228750389-d6355bd4-0c3e-4919-8339-de515101fd87.png)

Docker is a 'Shared folder' on Volume2 for me. So the local path to these folders would be;
-v /volume2/docker/frigate/... 
![image](https://user-images.githubusercontent.com/19168937/228751645-752d31a3-603b-4fb1-8569-b62ad336e5e2.png)
